yt/sample_data/api.py

```diff
@@ -0,0 +1,174 @@
+"""
+This is a collection of helper functions to yt.load_sample
+"""
+import json
+import re
+from functools import lru_cache
+from itertools import chain
+from pathlib import Path
+
+import pkg_resources
+import requests
```

Comment:
note to self: requests is still an optional dep, I need to update this with an on-demand import